freerdp: drop freerdp 2.x, move to by-name

LordGrimmauld opened the pull request:
The freerdp package name previuously pointed to the unmaintained freerdp version 2.x. Version 2.x is no longer used in nixpkgs.
It was never clear `freerdp` would be pinned to 2.x versions indefinitely.

This removes the pin on the freerdp package and moves the package to by-name pattern.


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

peterhoeg commented on pkgs/top-level/all-packages.nix:
```diff
@@ -13384,19 +13384,7 @@ with pkgs;
-    ;
-
-  freerdpUnstable = freerdp;
+  freerdp3 = freerdp;
```
this should also go to `aliases.nix`

LordGrimmauld replied:
`aliases.nix` complains about uses of a deprecated alias within nixpkgs. I can put this in aliases.nix, but then i need to treewide replace freerdp3 -> freerdp. If you are fine with that, sure

LordGrimmauld replied:
if this is something that should be done, i have no problem doing so in here

LordGrimmauld committed:
treewide: switch to unpinned freerdp

alyssais commented:
The commit ordering isn't ideal, because in the middle of the two commits we have packages depending on an alias.  Could it be structured as follows?

1. Point freerdp to 3
2. treewide to use unversioned freerdp
3. Remove freerdp2 and move freerdp3 to be an alias

LordGrimmauld commented:
> The commit ordering isn't ideal, because in the middle of the two commits we have packages depending on an alias. Could it be structured as follows?
> 
>     1. Point freerdp to 3
> 
>     2. treewide to use unversioned freerdp
> 
>     3. Remove freerdp2 and move freerdp3 to be an alias

dropping freerdp2 can happen whenever, nothing actually depends on it. But fixed the history to be bisectable

alyssais reviewed (approved):
Looks good but now has a conflict that needs to be resolved.

LordGrimmauld committed:
freerdp: drop freerdp 2.x, move to by-name

The freerdp package name previuously pointed to the unmaintained freerdp version 2.x.
Version 2.x is no longer used in nixpkgs.
It was never clear `freerdp` would be pinned to 2.x versions indefinitely.

This removes the pin on the freerdp package and moves the package to by-name pattern.

LordGrimmauld committed:
freerdp: deprecate aliases

LordGrimmauld commented:
Ok this *should* be fine now

LordGrimmauld commented:
krdp: https://github.com/NixOS/nixpkgs/pull/393686
krdc is a bit more complex, working on it

theCapypara commented:
Hello! I was just adding my app which requires freerdp2: https://github.com/NixOS/nixpkgs/pull/392984, this is now preventing it from compiling.

LordGrimmauld commented:
Right - so there are two breakages now? I guess i'll do a partial revert then....

drupol commented:
Working on a fix @ https://github.com/NixOS/nixpkgs/pull/407726

LordGrimmauld commented:
> Working on a fix @ #407726

Thanks, fixes are preferred over reintroducing a 1-year-old version of a library that had security fixes.
If it doesn't work out, #407731 is an option too, basically only needs undraft and merge. But i'd rather not if at all possible.

gbtb commented:
Hello. I'm considering the possibility to return freerdp version 2 to nixpkgs. Why the decision to remove v2 completely was made @LordGrimmauld  ? In my practice, xfreerdp v3 has a broken clipboard support on KDE6+Wayland, while xfreerdp v2 works perfectly fine.

gbtb commented:
I haven't reported this exact bug upstream yet.
Looking further into available options with freerdp3, it seems to me that for usage with Wayland sdl-freerdp binary is recommended.
However, according to https://github.com/FreeRDP/FreeRDP/issues/9399#issuecomment-1723630867 clipboard support is only available with SDL3. Nixpkgs package still uses SDL2 and I checked that clipboard share indeed doesn't work.
I think I'll try to build our package with SDL3 and see if it works.

LordGrimmauld commented:
We have https://github.com/NixOS/nixpkgs/pull/393673 to build freerdp against sdl3
